### PR TITLE
[10.x] Ordering Results When Using `find`

### PR DESCRIPTION
**Pull request by devajmeireles**
When we are working with the model using the method `find` we can retrieve a collection of results, such as:

```php
App\Models\People::find([10,5]);

/*
  0 => [
    "id" => 5
    // ...
  ]
  1 => [
    "id" => 10
    // ...
]*/
```

This PR aims to introduce the ability to **preserve the ordination that was given to the `find` method**:

```php
App\Models\People::find([10,5]);

/*
  0 => [
    "id" => 10
    // ...
  ]
  1 => [
    "id" => 5
    // ...
]*/
```

In the above example, I requested the `find` with the ordination of 10 and 5, so the first result is id 10 and after id 5.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -441,7 +441,15 @@ public function fromQuery($query, $bindings = [])
     public function find($id, $columns = ['*'])
     {
         if (is_array($id) || $id instanceof Arrayable) {
-            return $this->findMany($id, $columns);
+            $results = $this->findMany($id, $columns);
+
+            if ($results instanceof Arrayable) {
+                return (new Collection(
```
Comment (devajmeireles) [inline src/Illuminate/Database/Eloquent/Builder.php:447]: The `new Collection(...)` here was used to preserve the keys of the array from the result.

**Comment by devajmeireles** [comment]
The error with `tests / PHP 8.2` is not related to the code of this PR. I was unable to reproduce it locally.

**Comment by antonkomarev** [comment]
But why do we all need it? It will affect the performance for all the applications.

**Comment by devajmeireles** [comment]
> But why do we all need it? It will affect the performance for all the applications.

Hey, @antonkomarev ! Thanks for your reply.

According to my review, no significant memory increase was noticed. I did the test with a thousand units of modelers and these were the outputs:

1000 models **without** the code of this PR:

```shell
Max Used: 5.76 MB | Mem. Used: 45.84 KB
```

1000 models **with** the code of this PR:

```shell
Max Used: 5.82 MB | Mem. Used: 45.84 KB
```

As simple as it sounds, I believe it makes sense to get the list of models ordered the way it was requested, especially if we consider new Laravel users - they may have already expected to retrieve the models according to the given IDs to the `find` method and that did not happen.

**Comment by antonkomarev** [comment]
What is the use case for it? I've never expected to get results from the database in same order as WHERE conditions are provided. There is always `sort by` available.

**Comment by devajmeireles** [comment]
@antonkomarev @mfn ,

It's like PR changed Laravel completely. It's like PR makes using Laravel difficult and turns it into something really bad. Seriously, you need to think bigger. Laravel is a framework with over 200 million downloads - the world uses it. If something doesn't work for you, it doesn't mean it doesn't work for others. This is very disheartening. I agree that you should have opinions, but the point that something like this is going to transform Laravel's life is absurd.

**Comment by antonkomarev** [comment]
@devajmeireles no offense, I've just dropped my IMHO. I don't see any use cases why records should be sorted in the same order as ids were passed to query. You can share some to make a stronger decision point.

If it's a rare case, as @mfn noted, it is easily achievable in the userland.

**Pull request closed without merging**